Marlin/src/core/macros.h

```diff
@@ -60,6 +60,8 @@
 #define _O2          __attribute__((optimize("O2")))
 #define _O3          __attribute__((optimize("O3")))
 
+#define IS_CONSTEXPR(...) __builtin_constant_p(__VA_ARGS__) // Only valid solution with C++14. Should use std::is_constant_evaluated() in C++20 instead
```

Comment:
We've often wondered how much of the STL could Marlin use while remaining compatible with AVR + Arduino IDE and keeping the code small and fast. Code dealing with strings would be more concise if we had a good string class, for example, but we still want a usable Marlin build to fit in 64k.

Reply:
Yep. I've had a lot of trouble making use of very simple C++11 STL's construct like `enable_if` so I just imagine the nightmare it would be if we relied on C++17 or C++20 features. Hopefully the most useful construct of STL are usually only one-liner so it's not a big deal to reimplement them.

I've once written a compile-time `printf` code for ESP32 platform, so it'd indeed save a lot of binary space (but it takes longer to compile). However, I was not using STL code, but I was using C++17 features (like constexpr), so again, it'll not build on teensy. Maybe later when the toolchain is upgraded ?